(DOCSP-39513): Consolidate Create page (#3270)
mongodb/docs-realm: source/sdk/crud/create/create-object-types.txt

:ref:`sdks-object-models`.

You can create objects whose object type is managed by the database instance. 
For more information, refer to 
:ref:`sdks-configure-and-open-database` or 
:ref:`sdks-configure-and-open-synced-database`.

For more information about write transactions and the methods you can use to
create an object, refer to :ref:`sdks-create-object-methods`.

For information about creating specific property types, including special
types, collections, and relationship properties, refer to
:ref:`sdks-create-specific-property-types`.

Create Specific SDK Object Types
--------------------------------

Before you can create a new object and persist it to the database, you must 
:ref:`sdks-object-models`. Then, include that object type in your database
schema when you open the database.

.. important:: Object Types Must Be in Your Schema

   You can only write objects whose object type is included in the database
   schema. If you try to reference or write an object of an object type 
   that isn't in your schema, the SDK returns a schema validation error.

.. _sdks-create-realm-object:

Create a Realm Object
~~~~~~~~~~~~~~~~~~~~~

> The SDKs refer to the base object type as a Realm object. This is a legacy
> of the former product name, Realm Database.

Reviewer: ```suggestion
The SDK refers to the base object type as a Realm object. This identifies it as an object type that can be managed by 
the Realm Core database.
```
current wording reads a little negative? maybe. not sure that's the best word, but i feel like we can tweak this slightly to tie into our new product naming guidelines (i.e. Atlas Device SDK & Realm Core database)

      .. include:: /includes/api-details/csharp/crud/create-realm-object-description.rst
      
   .. tab::
      :tabid: dart

      .. include:: /includes/api-details/dart/crud/create-realm-object-description.rst

   .. tab::
      :tabid: java

      .. include:: /includes/api-details/java/crud/create-realm-object-description.rst

   .. tab::
      :tabid: java-kotlin

      .. include:: /includes/api-details/java/crud/create-realm-object-description.rst

   .. tab::
      :tabid: javascript

      .. include:: /includes/api-details/javascript/crud/create-realm-object-description.rst

   .. tab::
      :tabid: kotlin

      .. include:: /includes/api-details/kotlin/crud/create-realm-object-description.rst

   .. tab::
      :tabid: objectivec

      .. include:: /includes/api-details/objectivec/crud/create-realm-object-description.rst

   .. tab::
      :tabid: swift

      .. include:: /includes/api-details/swift/crud/create-realm-object-description.rst

   .. tab::
      :tabid: typescript

      .. include:: /includes/api-details/typescript/crud/create-realm-object-description.rst

.. include:: /includes/sdk-examples/crud/create-realm-object.rst

Model
`````

> For more information about modeling an object, refer to: 

Reviewer: ```suggestion
For more information about modeling an object, refer to
```

:ref:`sdks-object-models`.

> .. include:: /includes/sdk-examples/crud/create-realm-object-model.rst

Reviewer: fyi - we _should_ models for all of the Kotlin examples, so these _should_ be easy to pop in as part of the follow-up tickets

      
   .. tab::
      :tabid: dart

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: java

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: java-kotlin

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: javascript

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: kotlin

      .. include:: /includes/api-details/kotlin/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: objectivec

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: swift

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

   .. tab::
      :tabid: typescript

      .. include:: /includes/api-details/generic/crud/create-embedded-object-description.rst

.. include:: /includes/sdk-examples/crud/create-embedded-object.rst

Model
`````

> For more information about modeling an embedded object, refer to: 

Reviewer: ```suggestion
For more information about modeling an embedded object, refer to
```

      
   .. tab::
      :tabid: dart

      .. include:: /includes/api-details/dart/crud/create-asymmetric-object-description.rst

   .. tab::
      :tabid: java

      .. include:: /includes/api-details/java/crud/create-asymmetric-object-not-supported.rst

   .. tab::
      :tabid: java-kotlin

      .. include:: /includes/api-details/java/crud/create-asymmetric-object-not-supported.rst

   .. tab::
      :tabid: javascript

      .. include:: /includes/api-details/generic/crud/create-asymmetric-object-description.rst

   .. tab::
      :tabid: kotlin

      .. include:: /includes/api-details/kotlin/crud/create-asymmetric-object-description.rst

   .. tab::
      :tabid: objectivec

      .. include:: /includes/api-details/generic/crud/create-asymmetric-object-description.rst

   .. tab::
      :tabid: swift

      .. include:: /includes/api-details/generic/crud/create-asymmetric-object-description.rst

   .. tab::
      :tabid: typescript

      .. include:: /includes/api-details/generic/crud/create-asymmetric-object-description.rst

.. include:: /includes/sdk-examples/crud/create-asymmetric-object.rst

Model
`````

> For more information about modeling an asymmetric object, refer to: 

Reviewer: ```suggestion
For more information about modeling an asymmetric object, refer to
```

database like you would any other object type.

.. tabs-drivers::

   .. tab::
      :tabid: cpp-sdk

      .. include:: /includes/api-details/cpp/crud/create-geospatial-object-not-supported.rst

   .. tab::
      :tabid: csharp
      
   .. tab::
      :tabid: dart

   .. tab::
      :tabid: java

      .. include:: /includes/api-details/java/crud/create-geospatial-object-not-supported.rst

   .. tab::
      :tabid: java-kotlin

      .. include:: /includes/api-details/java/crud/create-geospatial-object-not-supported.rst

   .. tab::
      :tabid: javascript

      .. include:: /includes/api-details/generic/crud/create-geospatial-object-js-ts-description.rst

   .. tab::
      :tabid: kotlin

   .. tab::
      :tabid: objectivec

   .. tab::
      :tabid: swift

   .. tab::
      :tabid: typescript

      .. include:: /includes/api-details/generic/crud/create-geospatial-object-js-ts-description.rst

.. include:: /includes/sdk-examples/crud/create-geospatial-data-object.rst

The following image shows the results of creating these two company objects:

.. figure:: /images/geopoints.png
   :alt: 2 GeoPoints
   :width: 150
   :lightbox:

Model
`````

> .. include:: /includes/sdk-examples/crud/create-geospatial-data-model.rst

Reviewer: ```suggestion
For more information about modeling a geospatial object, refer to
:ref:`sdks-define-geospatial-object`.

.. include:: /includes/sdk-examples/crud/create-geospatial-data-model.rst
```
